src/cloudflare/main.go

```diff
@@ -43,23 +40,26 @@ func purgeWithData(jsonData interface{}) {
 		fmt.Println("WARNING: Not purging cloudflare cache since a key is not specified!")
 		return
 	}
+
 	url := "https://api.cloudflare.com/client/v4/zones/" + zone + "/purge_cache"
-	data, err := json.Marshal(jsonData)
+	req, err := util.JSONRequest(url, jsonData)
 	if err != nil {
-		fmt.Println("Cloudflare marshal error", err)
+		fmt.Println("Cloudflare error building request", err)
 		return
 	}
-	fmt.Println("Cloudflare purging data:", string(data))
-	req, err := http.NewRequest("POST", url, bytes.NewBuffer(data))
-	req.Header.Set("Authorization", "Bearer "+key)
-	req.Header.Set("Content-Type", "application/json")
+	req.Authorization("Bearer", key)
 
-	resp, err := (&http.Client{}).Do(req)
+	resp, err := req.Do()
 	if err != nil {
 		fmt.Println("Cloudflare purge error", err)
 		return
 	}
-	defer resp.Body.Close()
-	body, _ := ioutil.ReadAll(resp.Body)
-	fmt.Println("Cloudflare response body:", string(body))
+
+	body := resp.String()
+	if err != nil {
+		fmt.Println("Cloudflare error reading body", err)
```

Comment:
this err is no longer associated with body reading? since .String() is no longer fallible?

Reply:
Good catch, crept in with 2c11ff9